src/util/curl/curl-utils.ts

```diff
@@ -29,9 +30,11 @@ export const runCurl = async (opts: {
 			...process.env
 		}
 	});
-	let meta = stderr;
-	meta = _.replace(meta, new RegExp(_.escapeRegExp(opts.authorization), "g"), "******");
-	meta = _.replace(meta, new RegExp("set-cookie.+", "g"), "");
+	let meta = (cmd + "\n" + stderr).split(opts.authorization).join("********");
+	knownSensitiveHttpHeaders.forEach(header => {
+		// Some returned values might contain spaces/quotes
+		meta = _.replace(meta, new RegExp(`(< ${header}:\\s+?).*`, "gi"), "$1*******");
```

Comment:
what about the `knownSensitiveHttpHeaders` on the way out `> ${headers}` ?

Reply:
Is the `?` part of the `greedy` or not algorithm? Why do we need to require a <space> after the `${header}:`?

Reply:
`<` is the replied headers, no need to sanitize ">" (headers that we send) because we explicitly remove the authorization on the line above. Here's the output from the test

```curl -s -v  -H "Accept: application/json" -H "Authorization: ********" 'https://www.atlassian.com'
*   Trying 104.192.143.14:443...
* Connected to www.atlassian.com (104.192.143.14) port 443 (#0)
* ALPN: offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
} [322 bytes data]
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* (304) (IN), TLS handshake, Unknown (8):
{ [15 bytes data]
* (304) (IN), TLS handshake, Certificate (11):
{ [2818 bytes data]
* (304) (IN), TLS handshake, CERT verify (15):
{ [79 bytes data]
* (304) (IN), TLS handshake, Finished (20):
{ [52 bytes data]
* (304) (OUT), TLS handshake, Finished (20):
} [52 bytes data]
* SSL connection using TLSv1.3 / AEAD-AES256-GCM-SHA384
* ALPN: server accepted h2
* Server certificate:
*  subject: C=AU; ST=New South Wales; L=Sydney; O=Atlassian Pty Ltd; CN=*.atlassian.com
*  start date: May 23 00:00:00 2023 GMT
*  expire date: Jun 13 23:59:59 2024 GMT
*  subjectAltName: host "www.atlassian.com" matched cert's "*.atlassian.com"
*  issuer: C=US; O=DigiCert Inc; CN=DigiCert Global G2 TLS RSA SHA256 2020 CA1
*  SSL certificate verify ok.
* using HTTP/2
* h2h3 [:method: GET]
* h2h3 [:path: /]
* h2h3 [:scheme: https]
* h2h3 [:authority: www.atlassian.com]
* h2h3 [user-agent: curl/7.88.1]
* h2h3 [accept: application/json]
* h2h3 [authorization: ********]
* Using Stream ID: 1 (easy handle 0x7fb74300a800)
> GET / HTTP/2
> Host: www.atlassian.com
> user-agent: curl/7.88.1
> accept: application/json
> authorization: ********
> 
< HTTP/2 200 
< date: Wed, 28 Jun 2023 04:51:33 GMT
< content-type: text/html;charset=UTF-8
< content-length: 119502
< server: AtlassianEdge
< set-cookie: *******
< set-cookie: *******
< set-cookie: *******
< x-magnolia-registration: Registered
< pragma: no-cache
< cache-control: no-cache, no-store, must-revalidate, max-age=0
< last-modified: Tue, 27 Jun 2023 23:40:04 GMT
< x-frame-options: deny
< content-security-policy: frame-ancestors 'none';
< set-cookie: *******
< x-content-type-options: nosniff
< x-xss-protection: 1; mode=block
< atl-traceid: 861a04b781f8e5d8
< report-to: {"endpoints": [{"url": "https://dz8aopenkvv6s.cloudfront.net"}], "group": "endpoint-1", "include_subdomains": true, "max_age": 600}
< nel: {"failure_fraction": 0.001, "include_subdomains": true, "max_age": 600, "report_to": "endpoint-1"}
< strict-transport-security: max-age=63072000; preload
< vary: Accept-Encoding
< 
{ [15408 bytes data]
* Connection #0 to host www.atlassian.com left intact
```

As to `?`, I don't think it makes difference for this case, removing to remove the question ;)